packages/components/src/sites-table/sites-table-tab-panel.tsx

```diff
@@ -0,0 +1,40 @@
+import styled, { StyledComponent } from '@emotion/styled';
+import { TabPanel } from '@wordpress/components';
+
+export interface SitesTableTab extends Omit< TabPanel.Tab, 'title' > {
+	title: React.ReactChild;
+}
+
+export interface TabPanelProps extends Omit< TabPanel.Props, 'children' | 'tabs' > {
+	tabs: readonly SitesTableTab[];
+	children: ( tab: SitesTableTab ) => JSX.Element;
+}
+
+/**
+ * A wrapper around WordPress's <TabPanel> component.
+ *
+ * - Adjust style so it aligns with what we want in the Sites Dashboard
+ * - Allows arbitrary elements in the tab titles so that we can show site counts (this is already possible
+ *   with the WordPress component even though the documentation says only strings can be used).
+ */
+export const SitesTableTabPanel: StyledComponent< TabPanelProps > = styled( TabPanel )`
+	.components-tab-panel__tabs {
+		overflow-x: auto;
+	}
+
+	.components-tab-panel__tabs-item {
+		--wp-admin-theme-color: var( --studio-gray-100 );
+		color: var( --studio-gray-60 );
+		padding: 0;
+		margin-right: 24px;
+		font-size: 16px;
+		flex-shrink: 0;
+	}
+	.components-tab-panel__tabs-item:hover,
+	.components-tab-panel__tabs-item.is-active,
+	.components-tab-panel__tabs-item.is-active:focus,
+	.components-tab-panel__tabs-item:focus:not( :disabled ) {
+		box-shadow: inset 0 -2px 0 0 var( --wp-admin-theme-color );
+		color: var( --studio-gray-100 );
+	}
+` as any;
```

Comment:
Let's not roll with `any`s. All my fellas hate `any`s.

```diff
diff --git a/packages/components/src/sites-table/sites-table-tab-panel.tsx b/packages/components/src/sites-table/sites-table-tab-panel.tsx
index 1c9f788a38..c3625c02e5 100644
--- a/packages/components/src/sites-table/sites-table-tab-panel.tsx
+++ b/packages/components/src/sites-table/sites-table-tab-panel.tsx
@@ -1,11 +1,17 @@
-import styled, { StyledComponent } from '@emotion/styled';
+import styled from '@emotion/styled';
 import { TabPanel } from '@wordpress/components';
+import type { ComponentType } from 'react';
 
-export interface SitesTableTab extends Omit< TabPanel.Tab, 'title' > {
+type OmitFromIndexedTypesWhilePreservingProperties< T, K extends PropertyKey > = {
+	[ P in keyof T as Exclude< P, K > ]: T[ P ];
+};
+
+export interface SitesTableTab
+	extends OmitFromIndexedTypesWhilePreservingProperties< TabPanel.Tab, 'title' > {
 	title: React.ReactChild;
 }
 
-export interface TabPanelProps extends Omit< TabPanel.Props, 'children' | 'tabs' > {
+export interface SitesTabPanelProps extends Omit< TabPanel.Props, 'children' | 'tabs' > {
 	tabs: readonly SitesTableTab[];
 	children: ( tab: SitesTableTab ) => JSX.Element;
 }
@@ -17,7 +23,7 @@ export interface TabPanelProps extends Omit< TabPanel.Props, 'children' | 'tabs'
  * - Allows arbitrary elements in the tab titles so that we can show site counts (this is already possible
  *   with the WordPress component even though the documentation says only strings can be used).
  */
-export const SitesTableTabPanel: StyledComponent< TabPanelProps > = styled( TabPanel )`
+export const SitesTableTabPanel = styled( TabPanel as ComponentType< SitesTabPanelProps > )`
 	.components-tab-panel__tabs {
 		overflow-x: auto;
 	}
@@ -37,4 +43,4 @@ export const SitesTableTabPanel: StyledComponent< TabPanelProps > = styled( TabP
 		box-shadow: inset 0 -2px 0 0 var( --wp-admin-theme-color );
 		color: var( --studio-gray-100 );
 	}
-` as any;
+`;

```

We need this `OmitFromIndexedTypesWhilePreservingProperties` type because when `Omit` receives an object with property indexing ([the case of `TabPanel.Tab`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__components/tab-panel/index.d.ts#L21)), it gets confused and remove the required properties. More here: https://stackoverflow.com/questions/54826012/using-omit-type-on-classes-with-a-index-type-signature-results-in-minimum-proper.

Doing that, the types will actually start to overlap, so casting `TabPanel` as `ComponentType< SitesTabPanelProps >` will make TypeScript happy.

It will also fix the type in  `selectedTabName`. Now it's property typed as `string | undefined` instead of `any`. It wasn't because of the problem with `Omit` that was removing all the required properties and only considering the indexed type.

Reply:
TIL about that property indexing issue!

Personally I don't mind the use of `any` that much, especially when it really is being used to forcefully coerce types like we are here. To me, `any` can be a semantically correct type. So I guess I disagree that we should have a lint warning about it 😄 

In saying that, the `Omit` clearly wasn't working as I expected so I needed to fix that up anyway. And I think the way you did the type casting is much better.